src/Inventory.zig

```diff
@@ -349,11 +364,13 @@ pub const Sync = struct { // MARK: Sync
 			return inventories.items[serverId].inv;
 		}
 
-		pub fn getInventoryFromSource(source: SourceType) ?Inventory {
+		pub fn getUserInventoryFromSource(user: *main.server.User, source: SourceType) ?Inventory {
 			main.utils.assertLocked(&mutex);
 			for(inventories.items) |inv| {
-				if (inv.source == source) {
-					return inv.inv;
+				for (inv.users.items) |u| {
```

Comment:
You are just testing whether the player has openend this particular inventory, not whether it belongs to that player. Putting it into the `Source` type would be more robust.